compileopts/target.go

```diff
@@ -118,15 +118,15 @@ func (spec *TargetSpec) load(r io.Reader) error {
 
 // loadFromGivenStr loads the TargetSpec from the given string that could be:
 // - targets/ directory inside the compiler sources
-// - a relative or absolute path to custom (project specific) target specification .json file;
+// - a relative or absolute path to custom (project specific) target specification .yaml file;
 //   the Inherits[] could contain the files from target folder (ex. stm32f4disco)
-//   as well as path to custom files (ex. myAwesomeProject.json)
+//   as well as path to custom files (ex. myAwesomeProject.yaml)
 func (spec *TargetSpec) loadFromGivenStr(str string) error {
 	path := ""
-	if strings.HasSuffix(str, ".json") {
+	if strings.HasSuffix(str, ".yaml") {
```

Comment:
This comment applies to a lot of this change; I think we should do something like accept both `.json` and `.yaml` as valid extensions, rather than just completely switch from one to the other.

Reply:
I don't see a point to have mix of .json and .yaml files. That would introduce new code with risk of bugs that is doomed to be removed shortly after. And removal can also bring bugs.

Classic JSON is a subset of YAML and can be parsed by YAML parsers, so having .yaml extension for [classic] .json is not that bad actually.

Reformatting all target files -- that we can do. But since it will require curated, though automatic, conversion, I'd like to save this for the last step when everyone agrees we shall move to YAML and not something else, or not move at all.

I can make the conversion, but let's first decide we go this way at all?